manifests/resource/upstream.pp

```diff
@@ -105,8 +159,24 @@
   }
 
   concat::fragment { "${name}_upstream_footer":
-    target  => "${nginx::conf_dir}/${conf_dir_real}/${name}-upstream.conf",
+    target  => "${conf_dir}/${name}-upstream.conf",
     order   => '90',
-    content => template('nginx/upstream/upstream_footer.erb'),
+    content => epp('nginx/upstream/upstream_footer.epp', {
```

Comment:
My personal preference is for no parameters for `epp()`, where possible.

Reply:
I highly prefer them, because that allows scoped variables with proper datatypes.